packages/@aws-cdk-testing/cli-integ/lib/shell.ts

```diff
@@ -44,6 +47,25 @@ export async function shell(command: string[], options: ShellOptions = {}): Prom
         writeToOutputs(chunk);
       }
       stdout.push(chunk);
+
+      const interaction = interact.shift();
+      if (interaction) {
+
+        if (child.stdin == null) {
+          throw new Error('User interaction configured but child process has no stdin');
+        }
+
+        if (interaction.prompt.test(chunk)) {
```

Comment:
I don't think this is safe.

A chunk is going to be anywhere from 1..8192 bytes, and there's no guarantee a line will always be fully contained within a chunk. The following is conceivable:

```
this is a long line with a newline\nhere is another line\n
|-------|----------|--------|------------|----------|----|
```

Just because you haven't seen this on your machine while testing, doesn't mean it can't happen. And the testing code runs on many platforms under many conditions, and debugging them sucks, so I'd like this to be rock solid.

If you want to reliably match on lines, I think you're going to need to do accumulating and splitting yourself.

Reply:
Thanks, completely blanked on the fact a `chunk` is not necessarily a full line. I'll revise.

Reply:
Addressed